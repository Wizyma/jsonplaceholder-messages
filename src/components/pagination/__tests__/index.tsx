import React, { useEffect } from 'react';
import { render, cleanup } from '@testing-library/react';

import Component from '..';
import PaginationProvider, { usePaginationContext } from '../../../context/pagination-context';

beforeEach(() => {
  cleanup();
});

function WithPagination({ children }: { children: any }) {
  const { setCount } = usePaginationContext();

  useEffect(() => {
    setCount(10)
  }, [])

  return children
}

describe(`${Component.name} should pass all test`, () => {
  it('render an empty fragment', () => {
    const tree = (
      <PaginationProvider>
        <Component />
      </PaginationProvider>
    )
    const { asFragment } = render(tree)

    expect(asFragment()).toMatchSnapshot()
  })

  it('render a fragment with  the pagination', () => {
    const { asFragment } = render(
      <WithPagination>
        <Component />
      </WithPagination>, {
      wrapper: ({ children }) => (
        <PaginationProvider>
          {children}
        </PaginationProvider>
      )
    })

    expect(asFragment()).toMatchSnapshot()
  })
})