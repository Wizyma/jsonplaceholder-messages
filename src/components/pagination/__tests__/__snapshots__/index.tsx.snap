// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should pass all test render a fragment with  the pagination 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-2"
  >
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root"
    >
      <ul
        class="MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-selected"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            2
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 3"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            3
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 4"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            4
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 5"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            5
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <div
            class="MuiPaginationItem-root MuiPaginationItem-ellipsis"
          >
            â€¦
          </div>
        </li>
        <li>
          <button
            aria-label="Go to page 10"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            10
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Pagination should pass all test render an empty fragment 1`] = `<DocumentFragment />`;
